integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala

```diff
@@ -401,28 +396,21 @@ class CarbonTableCompactor(carbonLoadModel: CarbonLoadModel,
       carbonTable: CarbonTable,
       segments: Array[Segment]
   ): DataFrame = {
-    val columns = carbonTable
-          .getCreateOrderColumn()
-      .asScala
-      .map(_.getColName)
-      .toArray
-    val schema = SparkTypeConverter.createSparkSchema(carbonTable, columns)
-    val rdd: RDD[Row] = new CarbonScanRDD[CarbonRow](
-      sparkSession,
-      columnProjection = new CarbonProjection(columns),
-      null,
-      carbonTable.getAbsoluteTableIdentifier,
-      carbonTable.getTableInfo.serialize,
-      carbonTable.getTableInfo,
-      new CarbonInputMetrics,
-      null,
-      null,
-      classOf[CarbonRowReadSupport],
-      splitsOfSegments(sparkSession, carbonTable, segments))
-      .map { row =>
-        new GenericRow(row.getData.asInstanceOf[Array[Any]])
-      }
-    sparkSession.createDataFrame(rdd, schema)
+    CarbonSession
+      .threadSet(CarbonCommonConstants.CARBON_INPUT_SEGMENTS +
+                 carbonTable.getTableName + CarbonCommonConstants.POINT +
+                 carbonTable.getDatabaseName,
+        segments.map(s => s.getSegmentNo).mkString(","))
+    val logicalPlan = sparkSession
+      .sql(s"select * from ${ carbonTable.getTableName }")
+      .queryExecution
+      .logical
+    val dataFrame = PreAggregateUtil.getDataFrame(sparkSession, logicalPlan)
```

Comment:
Please move getDataFrame method to some CommonUtil

Reply:
done

Reply:
Done